lib/components/result-view/plotly.js

```diff
@@ -0,0 +1,149 @@
+/* @flow */
+/**
+ * Adapted from https://github.com/nteract/nteract/blob/master/packages/transform-plotly/src/index.tsx
+ * Copyright (c) 2016 - present, nteract contributors
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the LICENSE file in
+ * the root directory of this source tree.
+ *
+ * @NOTE: This `PlotlyTransform` component could be used exactly same as the original `PlotlyTransform` component of @nteract/transform-plotly,
+ *        except that this file adds the ability to download a plot from an electron context.
+ */
+
+import { cloneDeep } from "lodash";
+import * as React from "react";
+
+interface Props {
+  data: string | Object;
+  mediaType: "application/vnd.plotly.v1+json";
+}
+
+type ObjectType = Object;
+
+interface FigureLayout extends ObjectType {
+  height?: string;
+  autosize?: boolean;
+}
+
+interface Figure extends ObjectType {
+  data: Object;
+  layout: FigureLayout;
+}
+
+declare class PlotlyHTMLElement extends HTMLDivElement {
+  data: Object;
+  layout: Object;
+  newPlot: () => void;
+  redraw: () => void;
+}
+
+const NULL_MIMETYPE = "text/vnd.plotly.v1+html";
```

Comment:
We can remove the `NULL_MIMETYPE` stuff, I believe.

Reply:
Ok will do